From ccb1cc57c6cfdb5952c3b67a620609a0a37d1425 Mon Sep 17 00:00:00 2001
From: famfo <famfo@famfo.xyz>
Date: Mon, 13 Jan 2025 19:31:58 +0100
Subject: [PATCH] remove statically linked deps, remove ANSI support

- removes ANSI support from tracing-subscriber for
  the server log
- removes rustls from reqwest and sqlx in favor of
  dynamically linked system libraries

Signed-off-by: famfo <famfo@famfo.xyz>
---
 Cargo.toml |   7 +-
 2 files changed, 110 insertions(+), 104 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index f54652c9..164e2c06 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -45,16 +45,17 @@ tracing = "0.1"
 
 [workspace.dependencies.tracing-subscriber]
 version = "0.3"
-features = ["ansi", "fmt", "registry", "env-filter"]
+default-features = false
+features = ["fmt", "registry", "env-filter", "smallvec", "std", "tracing-log"]
 
 [workspace.dependencies.reqwest]
 version = "0.11"
-features = ["json", "rustls-tls-native-roots"]
+features = ["json", "native-tls"]
 default-features = false
 
 [workspace.dependencies.sqlx]
 version = "0.8"
-features = ["runtime-tokio-rustls", "time", "postgres", "uuid"]
+features = ["runtime-tokio-native-tls", "time", "postgres", "uuid"]
 
 # The profile that 'cargo dist' will build with
 [profile.dist]
-- 
2.48.0

